kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -32,7 +34,8 @@
  * @param filters filers.
  */
 @SuppressWarnings("java:S1123") // suppressing the spurious warning about missing @deprecated in javadoc. It is the field that is deprecated, not the class.
-public record VirtualCluster(TargetCluster targetCluster,
+public record VirtualCluster(@NonNull @JsonProperty(required = true) String name,
+                             @NonNull @JsonProperty(required = true) TargetCluster targetCluster,
```

Comment:
Is the `required` property unit tested for the two fields? Would be good to have unit tests checking null/absent in JSON provokes an error.